web-ui/src/pages/CheckinsReportPage.jsx

```diff
@@ -97,7 +97,7 @@ const CheckinsReportPage = () => {
           const newPdls = ids
             .map(id => pdls.find(pdl => pdl.id === id))
             .filter(Boolean);
-          setSelectedPdls(newPdls);
+          newPdls.length > 0 && setSelectedPdls(newPdls);
```

Comment:
If `newPdls` becomes empty, does it to clear out the array of selected PDLs?

Reply:
For `newPdls` to become empty I believe we would have a data issue on our hands. This change is to allow us to short-circuit the logic for changing component state as the default state when no pdls are available is controlled in a `useEffect` queuing off `processedQPs.current`.

Reply:
I see what you're saying. This looks good. I actually didn't noticed that this was in code related to the query parameters. Makes sense now.